x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/import/action_connectors/import_rule_action_connectors.test.ts

```diff
@@ -380,4 +380,252 @@ describe('checkRuleExceptionReferences', () => {
+        meta: { title: 'Connector: [anotherSpaceSlack]', icon: undefined },
+        type: 'action',
+      },
+    ];
+    core.savedObjects.getImporter = jest.fn().mockReturnValueOnce({
+      import: jest.fn().mockResolvedValue({
+        success: true,
+        successCount: 1,
+        successResults,
+        errors: [],
+        warnings: [],
+      }),
+    });
+    const actionsImporter = core.savedObjects.getImporter;
+
+    actionsClient.getAll.mockResolvedValue([]);
+
+    const res = await importRuleActionConnectors({
+      actionConnectors,
+      actionsClient,
+      actionsImporter: actionsImporter() as never,
+      rules,
+      overwrite: false,
+    });
+    const rulesWithMigratedActions = [
+      {
+        actions: [
+          {
+            action_type_id: '.webhook',
+            group: 'default',
+            id: '72cab9bb-535f-45dd-b9c2-5bc1bc0db96b',
+            params: {},
+          },
+        ],
+        description: 'some description',
+        language: 'kuery',
+        name: 'Query with a rule id',
+        query: 'user.name: root or user.name: admin',
+        risk_score: 55,
+        rule_id: 'rule-1',
+        severity: 'high',
+        type: 'query',
+      },
+    ];
+
+    expect(res).toEqual({
+      success: true,
+      successCount: 1,
+      errors: [],
+      warnings: [],
+      successResults,
+      rulesWithMigratedActions,
+    });
+  });
+  it('should generate new destinationIds if the connectors are exported from a different namespace', async () => {
+    const successResults = [
+      {
+        destinationId: '72cab9bb-535f-45dd-b9c2-5bc1bc0db96b',
+        id: 'cabc78e0-9031-11ed-b076-53cc4d57aaf1',
+        meta: { title: 'Connector: [anotherSpaceSlack]', icon: undefined },
+        type: 'action',
+      },
+    ];
+    core.savedObjects.getImporter = jest.fn().mockReturnValueOnce({
+      import: jest.fn().mockResolvedValue({
+        success: true,
+        successCount: 1,
+        successResults,
+        errors: [],
+        warnings: [],
+      }),
+    });
+    const actionsImporter = core.savedObjects.getImporter;
+
+    actionsClient.getAll.mockResolvedValue([]);
+
+    const res = await importRuleActionConnectors({
+      actionConnectors,
+      actionsClient,
+      actionsImporter: actionsImporter() as never,
+      rules,
+      overwrite: false,
+    });
+    const rulesWithMigratedActions = [
+      {
+        actions: [
+          {
+            action_type_id: '.webhook',
+            group: 'default',
+            id: '72cab9bb-535f-45dd-b9c2-5bc1bc0db96b',
+            params: {},
+          },
+        ],
+        description: 'some description',
+        language: 'kuery',
+        name: 'Query with a rule id',
+        query: 'user.name: root or user.name: admin',
+        risk_score: 55,
+        rule_id: 'rule-1',
+        severity: 'high',
+        type: 'query',
+      },
+    ];
+
+    expect(res).toEqual({
```

Comment:
Ideally we need to avoid test flakiness so we basically need to check whether we replaced action ids with destination ids here so `jest.objectContaining` should do the trick help to make the test much more readable.

Reply:
Here I am not only checking the replacement of the actionids I need to check the whole results which include changing the right action id of the right rule as well as returning from the `importRuleActionConnectors` with the expected results.
I am not sure, why this will cause flakiness. please share with me, what you see 🙏🏻

Reply:
Please check if the test description is more relevant to the test or not, thank you! [6d457dc](https://github.com/elastic/kibana/pull/151671/commits/6d457dc1c3ed961f1f4b9418a506157519ba1c2a)